src/bundles/peer-locations.js

```diff
@@ -250,26 +275,23 @@ class PeerLocationResolver {
   }
 
   optimizedPeerSet (peers) {
-    if (this.pass && this.pass < 3) {
+    if (this.pass < 3) {
       // use a copy of peers sorted by latency so we can resolve closest ones first
       // (https://github.com/ipfs-shipyard/ipfs-webui/issues/1273)
       const ms = x => (parseLatency(x.latency) || 9999)
-      peers = peers.concat().sort((a, b) => ms(a) - ms(b))
+      const sortedPeersByLatency = peers.concat().sort((a, b) => ms(a) - ms(b))
```

Comment:
do we need to copy if we are going to use slice below ?

Reply:
For the sake of immutability, `sort` does affect the original array, so it would be best to.

Thoughts on this? 🤔

Reply:
if we sort it once upon creation and work with it sorted we should be safe.

Reply:
I guess you're right, I'll change it 😄

Reply:
After further analysis, I just realized that this is the poller and having this sort in this specific part, actually reduces the number of times the array gets sorted since it only does it for the 3 first times.